py/src/braintrust/logger.py

```diff
@@ -88,12 +88,15 @@ def log_feedback(self, **event):
         """
 
     @abstractmethod
-    def start_span(self, name=None, span_attributes={}, start_time=None, set_current=None, parent_id=None, **event):
+    def start_span(
+        self, name=None, type=None, span_attributes={}, start_time=None, set_current=None, parent_id=None, **event
+    ):
         """Create a new span. This is useful if you want to log more detailed trace information beyond the scope of a single log event. Data logged over several calls to `Span.log` will be merged into one logical row.
 
         We recommend running spans within context managers (`with start_span(...) as span`) to automatically mark them as current and ensure they are ended. Only spans run within a context manager will be marked current, so they can be accessed using `braintrust.current_span()`. If you wish to start a span outside a context manager, be sure to end it with `span.end()`.
 
         :param name: Optional name of the span. If not provided, a name will be inferred from the call stack.
+        :param type: Optional type of the span. If not provided, the type will be unset.
```

Comment:
The type is a bit unclear to me. Is it supposed to be an enum or a string? Seems like enum, but in typescript we allow strings. In python there is a `Literal` type that could annotate that it must be of a subset of strings, and then we could convert that to the enum / log a warning.

Reply:
added a clearer message. We should probably add type annotations to all the arguments here.